feature_engine/transformation/ciclycal.py

```diff
@@ -0,0 +1,122 @@
+
+
+class CyclicalTransformer(BaseNumericalTransformer):
+    """
+    The CyclicalTransformer() applies a Ciclycal transformation to numerical
+    variables.
+
+    There are some feature that are cyclic by nature. One example of this are
+    the hours of a day or the months of a year. In both cases the higher values of
+    a set of data are closer to the lower values of that set.
+
+    For example the month October (10) is closer to month January (1) than
+    to month February (2).
+
+    To check an explanation about this:
+    http://blog.davidkaleko.com/feature-engineering-cyclical-features.html
+
+    The CyclicalTransformer() works only with numerical variables. But does not
+    allow null values on any row of original column
+
+    A list of variables can be passed as an argument. Alternatively, the
+    transformer will automatically select and transform all numerical
+    variables.
+
+    Parameters
+    ----------
+    variables : list, default=None
+        The list of numerical variables that will be transformed. If None, the
+        transformer will automatically find and select all numerical variables.
+
+    Methods
+    -------
+    fit:
+        This transformer does not learn parameters.
+    transform:
+        Apply the CyclicalTransformer transformation.
+    fit_transform:
+        Fit to data, then transform it.
+
+    References
+    ----------
+    ..
+    """
+
+    def __init__(
+            self, variables: Union[None, int, str, List[Union[str, int]]] = None
+    ) -> None:
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn any parameter. But keeps a map of the max values
+        for every column
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y : pandas Series, default=None
+            It is not needed in this transformer. You can pass y or None.
+
+        Raises
+        ------
+        TypeError
+            If the input is not a Pandas DataFrame.
+        ValueError:
+            If some of the columns contains NaNs
+
+        Returns
+        -------
+        self
+        """
+
+        # check input dataframe
+        X = super().fit(X)
+
+        self.input_shape_ = X.shape
+
+        # check for nans
+        self.max_values_ = {}
+        for variable in self.variables:
+            if X[variable].isna().sum() > 0:
+                raise ValueError(f'The transformer CiclycalTransformer '
+                                 f'does not allow to have NaN values, please '
+                                 f'check the column {variable}')
+            self.max_values_[variable] = X[variable].max()
```

Comment:
Instead of the loop, could we change the presence of NA, with a feature engine function: 
_`check_contains_na(X, self.variables)`?

Also, can we make a dictionary with 1 line instead:
`self.max_values_ = X[self.variables].max().to_dict()` ?

Finally, I see in the article that it substracts 1 to months because their scale is 1-12, but it does not subtract 1 to hr, because they are already starting in 0, so 0-23, instead of 1-24. 

How does this affect the cyclical transformation?

Should we not also check for the minimum value and if it is not 0, subtract to reindex in 0?

Reply:
I will check this. Not sure how does it affects the ciclycal transformation.